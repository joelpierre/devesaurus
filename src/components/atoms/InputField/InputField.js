import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './InputField.module.scss';
import Label from '../Label/Label';

const InputField = ({
  name,
  label,
  type,
  placeholder,
  disabled,
  readonly,
  min,
  max,
  sm,
  md,
  lg,
  defaultValue,
  hidden,
  onChange,
  onBlur,
  onFocus,
  onClick,
}) => {
  let inputField;
  switch (type) {
    case 'text':
    case 'email':
    case 'password':
    case 'date':
      inputField = (
        <input
          data-test="component-input-field"
          className={classNames({
            [styles['input-field']]: true,
            [styles['input-field--sm']]: sm,
            [styles['input-field--md']]: md,
            [styles['input-field--lg']]: lg,
          })}
          type={type}
          name={name}
          placeholder={placeholder}
          readOnly={readonly}
          disabled={disabled}
          defaultValue={defaultValue}
          aria-label={`Input for ${name}`}
          hidden={hidden}
          onChange={onChange}
          onBlur={onBlur}
          onFocus={onFocus}
          onClick={onClick}
        />
      );
      break;
    case 'number':
      inputField = (
        <input
          data-test="component-input-field"
          className={classNames({
            [styles['input-field']]: true,
            [styles['input-field--sm']]: sm,
            [styles['input-field--md']]: md,
            [styles['input-field--lg']]: lg,
          })}
          type={type}
          name={name}
          minLength={min}
          maxLength={max}
          placeholder={placeholder}
          readOnly={readonly}
          disabled={disabled}
          defaultValue={defaultValue}
          aria-label={`Input for ${name}`}
          hidden={hidden}
          onChange={onChange}
          onBlur={onBlur}
          onFocus={onFocus}
          onClick={onClick}
        />
      );
      break;
    default:
      inputField = (
        <input
          data-test="component-input-field"
          className={classNames({
            [styles['input-field']]: true,
            [styles['input-field--sm']]: sm,
            [styles['input-field--md']]: md,
            [styles['input-field--lg']]: lg,
          })}
          type="text"
          name={name}
          placeholder={placeholder}
          readOnly={readonly}
          disabled={disabled}
          defaultValue={defaultValue}
          aria-label={`Input for ${name}`}
          hidden={hidden}
          onChange={onChange}
          onBlur={onBlur}
          onFocus={onFocus}
          onClick={onClick}
        />
      );
  }
  return (
    <>
      {label && <Label text={label} name={name} />}
      {inputField}
    </>
  );
};

InputField.defaultProps = {
  placeholder: '',
  disabled: false,
  readonly: false,
  min: null,
  max: null,
  sm: false,
  md: false,
  lg: false,
  label: null,
  defaultValue: '',
  hidden: null,
  onChange: null,
  onBlur: null,
  onFocus: null,
  onClick: null,
};

InputField.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string,
  type: PropTypes.oneOf([
    'text',
    'file',
    'select',
    'password',
    'date',
    'email',
    'number',
  ]).isRequired,
  placeholder: PropTypes.string,
  disabled: PropTypes.bool,
  readonly: PropTypes.bool,
  sm: PropTypes.bool,
  md: PropTypes.bool,
  lg: PropTypes.bool,
  min: PropTypes.string,
  max: PropTypes.number,
  defaultValue: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  hidden: PropTypes.bool,
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
  onFocus: PropTypes.func,
  onClick: PropTypes.func,
};

export default InputField;
