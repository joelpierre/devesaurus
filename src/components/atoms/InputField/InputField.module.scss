@import "~sass/core";

.input-field {
  width: 100%;
  border-radius: $radius-rounded;
  border: 1px solid $color-tint-gamma;
  background-color: white;
  line-height: 1.2;
  padding: 8px 16px;
  height: 40px;
  margin-bottom: 2rem;

  &--sm {
    height: 30px;
    padding: 2px 10px;
  }

  &--md {
    height: 40px;
    padding: 8px 16px;
  }

  &--lg {
    height: 50px;
    padding: 10px 20px;
  }

  &[type='text'] {
    appearance: none;
  }

  &[type='file'] {
    border: none;
    padding-left: 0;
    padding-right: 0;
  }

  &[type='radio'],
  &[type='checkbox'] {
    height: auto;
  }
}

.select-wrap {
  position: relative;
  display: inline-block;
}
