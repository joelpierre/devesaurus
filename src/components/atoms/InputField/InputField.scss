@import "~sass/core";

.form-control {

}

.select-wrap {
  position: relative;
  display: inline-block;
}
