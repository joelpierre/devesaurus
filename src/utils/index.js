export const updateObject = (oldObject, updatedValues) => ({
  ...oldObject,
  ...updatedValues,
});

export const mapOverACFComponents = components => {
  components.map(component => {
    component.acf_fc_layout = component.acf_fc_layout.replace(
      'WordPressAcf_',
      ''
    );
  });
  return components;
};

export const sortWordObj = word => {
  if (word) {
    delete word.author_id;
    delete word.author;
    delete word.author_nicename;
    delete word.content;
    delete word.date_modified;
    delete word.excerpt;
    delete word.yoast;

    return word;
  }

  return false;
};

export const randTheme = () => {
  const theme = ['brand', 'alpha', 'beta', 'gamma', 'psi', 'omega'];
  const rand = randNumBetween(0, theme.length);
  return theme[rand];
};

export const randNumBetween = (min = 0, max = 10) => {
  return Math.floor(Math.random() * max) + min;
};

export const mapTaxonomyTheme = (slug = 'default') => {
  let theme;

  switch (slug) {
    case 'development':
    case 'web-development':
      theme = 'alpha';
      break;
    case 'graphic-design':
    case 'design':
      theme = 'beta';
      break;
    case 'css':
    case 'css3':
      theme = 'gamma';
      break;
    case 'html':
    case 'html5':
      theme = 'psi';
      break;
    case 'news':
      theme = 'omega';
      break;
    default:
      theme = 'brand';
  }

  return theme;
};

export const breakpoint = {
  is(s) {
    const size = s.trim();
    const sizes = {
      xsmall: '599px',
      small: '600px',
      medium: '900px',
      large: '1200px',
      xlarge: '1800px',
    };

    // console.log(window.matchMedia);

    // Eslint doesn't like you accessing hasOwnProperty directly on object.
    // https://eslint.org/docs/rules/no-prototype-builtins
    if (Object.prototype.hasOwnProperty.call(sizes, size)) {
      return window.matchMedia(`only screen and (min-width: ${sizes[size]})`)
        .matches;
    }

    throw new ReferenceError(
      `The size ${size} is not a valid breakpoint: ${JSON.stringify(sizes)}`
    );
  },
};
