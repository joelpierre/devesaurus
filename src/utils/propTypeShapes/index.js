import PropTypes from 'prop-types';

export const wordPropTypeShape = {
  title: PropTypes.string.isRequired,
  slug: PropTypes.string.isRequired,
  terms: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      slug: PropTypes.string.isRequired,
      taxonomy: PropTypes.string.isRequired,
    })
  ),
  acf: PropTypes.shape({
    definition: PropTypes.string,
    origin: PropTypes.shape({
      label: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired,
    }),
    syllables: PropTypes.shape({
      count: PropTypes.string.isRequired,
      list: PropTypes.arrayOf(
        PropTypes.shape({
          item: PropTypes.string,
        })
      ).isRequired,
    }).isRequired,
    pronunciation: PropTypes.string,
  }),
};

export const themePropTypeShape = {
  theme: PropTypes.oneOf([
    'brand',
    'alpha',
    'tint-alpha',
    'tint-beta',
    'tint-psi',
    'tint-omega',
  ]),
};
