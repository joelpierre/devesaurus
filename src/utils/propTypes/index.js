import PropTypes from 'prop-types';

export const wordPropTypeShape = {
  title: PropTypes.string.isRequired,
  slug: PropTypes.string.isRequired,
  terms: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      slug: PropTypes.string.isRequired,
      taxonomy: PropTypes.string.isRequired,
    })
  ),
  acf: PropTypes.shape({
    definition: PropTypes.string,
    origin: PropTypes.shape({
      label: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired,
    }),
    syllables: PropTypes.shape({
      count: PropTypes.string.isRequired,
      list: PropTypes.arrayOf(
        PropTypes.shape({
          item: PropTypes.string,
        })
      ).isRequired,
    }).isRequired,
    pronunciation: PropTypes.string,
  }),
};

export const columnShape = PropTypes.oneOf([
  '1',
  1,
  '2',
  2,
  '3',
  3,
  '4',
  4,
  '5',
  5,
  '6',
  6,
  '7',
  7,
  '8',
  8,
  '9',
  9,
  '10',
  10,
  '11',
  11,
  '12',
  12,
]);

export const defaultSize = (size = 'md') => ({
  size,
});

export const sizePropType = {
  size: PropTypes.oneOf(['xs', 'sm', 'md', 'lg', 'xl', 'full']),
};

export const defaultTheme = (theme = 'brand') => ({
  theme,
});

export const themePropType = {
  theme: PropTypes.oneOf([
    'brand',
    'alpha',
    'beta',
    'gamma',
    'psi',
    'omega',
    'tint-alpha',
    'tint-beta',
    'tint-gamma',
    'tint-psi',
    'tint-omega',
    'gradient-brand',
  ]),
};

export const defaultPageTheme = (theme = 'brand') => ({
  pageTheme: theme,
});

export const pageThemePropType = {
  pageTheme: PropTypes.oneOf([
    'brand',
    'alpha',
    'beta',
    'gamma',
    'psi',
    'omega',
    'tint-alpha',
    'tint-beta',
    'tint-gamma',
    'tint-psi',
    'tint-omega',
  ]),
};
