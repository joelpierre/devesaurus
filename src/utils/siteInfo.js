import React from 'react';
import { graphql, StaticQuery } from 'gatsby';

const SiteInfo = () => (
  <StaticQuery
    query={graphql`
      {
        allWordpressSiteMetadata {
          edges {
            node {
              name
              description
            }
          }
        }
      }
    `}
    render={({ allWordpressSiteMetadata }) => (
      <>
        {allWordpressSiteMetadata.edges[0].node.name}
        <span>{allWordpressSiteMetadata.edges[0].node.description}</span>
      </>
    )}
  />
);

export default SiteInfo;
