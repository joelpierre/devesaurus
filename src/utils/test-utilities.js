import checkPropTypes from 'check-prop-types';
import { applyMiddleware, createStore } from 'redux';

import rootReducer from '../store/reducers';
// import { sagaMiddleware } from '../store/configureStore';

// /**
//  * Store Factory
//  * @param initialState
//  * @returns {Store<any, Action> & *}
//  */
// export const storeFactory = (initialState) => {
//   const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
//   return createStoreWithMiddleware(rootReducer, initialState);
// };


/**
 * Return node(s) with the given data-test attribute.
 * @param {ShallowWrapper} wrapper - Enzyme shallow wrapper.
 * @param {string} val - Value of the data-test attribute.
 * @returns {ShallowWrapper}
 */
export const findByTestAttr = (wrapper, val) => {
  return wrapper.find(`[data-test="${val}"]`);
};


/**
 * Check the props for a component
 * @param component
 * @param conformingProps
 */
export const checkProps = (component, conformingProps) => {
  const propError = checkPropTypes(
    component.propTypes,
    conformingProps,
    'prop',
    component.name,
  );
  expect(propError)
    .toBeUndefined();
};
