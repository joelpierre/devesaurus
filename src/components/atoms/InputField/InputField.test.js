import React from 'react';
import { shallow } from 'enzyme';

import InputField from './InputField';
import { findByTestAttr } from '../../../utils/test-utilities';

const defaultProps = {
  name: 'test-name',
  type: 'text',
};

/**
 * Factory function to create a ShallowWrapper for the InputField component.
 * @param {object} props - Component props specific to setup
 * @returns {ShallowWrapper}
 */
const setup = (props = {}) => {
  const setupProps = { ...defaultProps, ...props };
  return shallow(<InputField {...setupProps} />);
};

describe('<InputField/>', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = setup();
  });

  it('Renders the InputField Component without errors', () => {
    const component = findByTestAttr(wrapper, 'component-input-field');
    expect(component.length).toBe(1);
  });
});
