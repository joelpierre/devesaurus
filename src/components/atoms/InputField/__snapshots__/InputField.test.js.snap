// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputField/> should match snapshot 1`] = `
<Fragment>
  <input
    aria-label="Input for test"
    className="input-field"
    data-test="component-input-field"
    defaultValue=""
    disabled={false}
    hidden={null}
    name="test"
    onBlur={null}
    onChange={null}
    onClick={null}
    onFocus={null}
    placeholder=""
    readOnly={false}
    type="text"
  />
</Fragment>
`;
